packages/record/src/record/record-login-flow.ts

```diff
@@ -0,0 +1,214 @@
+import {
+  defer,
+  METICULOUS_LOGGER_NAME,
+  RecordLoginFlowSessionFn,
+} from "@alwaysmeticulous/common";
+import chalk from "chalk";
+import log from "loglevel";
+import { Browser, launch } from "puppeteer";
+import {
+  DEFAULT_NAVIGATION_TIMEOUT_MS,
+  DEFAULT_UPLOAD_INTERVAL_MS,
+  INITIAL_METICULOUS_RECORD_LOGIN_FLOW_DOCS_URL,
+  METICULOUS_RECORD_LOGIN_FLOW_SAVING_DOCS_URL,
+} from "./constants";
+import { bootstrapPage } from "./record.utils";
+import { exposeNewRecordingCallback } from "./utils/expose-new-recording-callback";
+import { injectFinishRecordingFrame } from "./utils/inject-finish-recording-frame";
+import {
+  printNoLoginSessionRecordedWarning,
+  printPageClosedWarning,
+} from "./utils/print-warning";
+
+type ModifiedWindow = {
+  __meticulous?: {
+    flushPendingPayloads?: () => Promise<void>;
+  };
+};
+
+export const LOGIN_FLOW_SESSION_RECORDING_SOURCE = "cli-login-flow";
+export const LOGIN_FLOW_DATA_SESSION_RECORDING_SOURCE = "cli-login-flow-data";
```

Comment:
I've been naming it 'application-storage' throughout (which is what Chrome collectively calls cookies + local storage + session storage etc.). So could do `cli-login-flow-application-storage`. `cli-login-flow-data` is fine too.